[FLINK-15192][docs][table] Restructure "SQL" pages for better readability (#10669)
apache/flink: docs/dev/table/sql/queries.md

In order to access a table in a SQL query, it must be [registered in the TableEnvironment]({{ site.baseurl }}/dev/table/common.html#register-tables-in-the-catalog). A table can be registered from a [TableSource]({{ site.baseurl }}/dev/table/common.html#register-a-tablesource), [Table]({{ site.baseurl }}/dev/table/common.html#register-a-table), [CREATE TABLE statement](#create-table), [DataStream, or DataSet]({{ site.baseurl }}/dev/table/common.html#register-a-datastream-or-dataset-as-table). Alternatively, users can also [register catalogs in a TableEnvironment]({{ site.baseurl }}/dev/table/catalogs.html) to specify the location of the data sources.

> For convenience `Table.toString()` automatically registers the table under a unique name in its `TableEnvironment` and returns the name. Hence, `Table` objects can be directly inlined into SQL queries (by string concatenation) as shown in the examples below.

Reviewer: ```suggestion
For convenience, `Table.toString()` automatically registers the table under a unique name in its `TableEnvironment` and returns the name. So, `Table` objects can be directly inlined into SQL queries as shown in the examples below.
```

> **Note:** Flink's SQL support is not yet feature complete. Queries that include unsupported SQL features cause a `TableException`. The supported features of SQL on batch and streaming tables are listed in the following sections.

Reviewer: I'm wondering if this first sentence is still necessary for if 1.10 is feature complete enough to remove it. I don't want potential users to become disuaded when we cover so much of the standard at this point. 

```suggestion
**Note:** Queries that include unsupported SQL features cause a `TableException`. The supported features of SQL on batch and streaming tables are listed in the following sections.
```

Author: +1, I also want to remove the first sentense. Because all the features listed in this page are supported by both planners. Compared to standard SQL or other systems, Flink is not fully feature complete. But we don't need to mention that.

{% top %}

> ## Supported Syntax

Author: I think syntax is a summary of what is supported and is an entrypoint of the features. Almost all the system are displaying syntax at the top of page, e.g. [SQL Server](https://docs.microsoft.com/en-us/sql/t-sql/queries/select-transact-sql), [Spark](https://docs.databricks.com/spark/latest/spark-sql/language-manual/select.html), [Hive](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Select). 

If users are interested in the operations, they are jump to the detailed section by TOC.
